Resolve fatal "No core keyboard" error on !x86 platforms (LP: #281610)

In dix/devices.c cast dev->enabled to Bool not CARD8. On big endian systems
using *(CARD8*)(&dev->enabled) doesn't look at the right byte. Bool is
typedef'd to int which produces the correct cross platform representation.

Patch taken from upstream bug 18111:
https://bugs.freedesktop.org/show_bug.cgi?id=18111

Index: xorg-server/dix/devices.c
===================================================================
--- xorg-server.orig/dix/devices.c	2008-10-23 07:15:14.000000000 +0100
+++ xorg-server/dix/devices.c	2008-10-23 07:21:47.000000000 +0100
@@ -99,12 +99,13 @@
 {
     if (property == XIGetKnownProperty(XI_PROP_ENABLED))
     {
-        if (prop->format != 8 || prop->type != XA_INTEGER || prop->size != 1)
+        if (prop->format != 8 * sizeof(dev->enabled) ||
+            prop->type != XA_INTEGER || prop->size != 1)
             return BadValue;
 
-        if ((*((CARD8*)prop->data)) && !dev->enabled)
+        if ((*((Bool*)prop->data)) && !dev->enabled)
             EnableDevice(dev);
-        else if (!(*((CARD8*)prop->data)) && dev->enabled)
+        else if (!(*((Bool*)prop->data)) && dev->enabled)
             DisableDevice(dev);
     }
 
@@ -200,8 +201,8 @@
     dev->next = NULL;
 
     XIChangeDeviceProperty(dev, XIGetKnownProperty(XI_PROP_ENABLED),
-                           XA_INTEGER, 8, PropModeReplace, 1, &dev->enabled,
-                           FALSE);
+                           XA_INTEGER, 8 * sizeof(dev->enabled),
+                           PropModeReplace, 1, &dev->enabled, FALSE);
     XISetDevicePropertyDeletable(dev, XIGetKnownProperty(XI_PROP_ENABLED), FALSE);
     XIRegisterPropertyHandler(dev, DeviceSetProperty, NULL, NULL);
 
@@ -243,8 +244,8 @@
     dev->next = NULL;
 
     XIChangeDeviceProperty(dev, XIGetKnownProperty(XI_PROP_ENABLED),
-                           XA_INTEGER, 8, PropModeReplace, 1, &dev->enabled,
-                           TRUE);
+                           XA_INTEGER, 8 * sizeof(dev->enabled),
+                           PropModeReplace, 1, &dev->enabled, TRUE);
 
     ev.type = DevicePresenceNotify;
     ev.time = currentTime.milliseconds;
@@ -284,8 +285,8 @@
     inputInfo.off_devices = dev;
 
     XIChangeDeviceProperty(dev, XIGetKnownProperty(XI_PROP_ENABLED),
-                           XA_INTEGER, 8, PropModeReplace, 1, &dev->enabled,
-                           TRUE);
+                           XA_INTEGER, 8 * sizeof(dev->enabled),
+                           PropModeReplace, 1, &dev->enabled, TRUE);
 
     ev.type = DevicePresenceNotify;
     ev.time = currentTime.milliseconds;
